Hakawai/HKWMentionsPluginV2.m

```diff
@@ -0,0 +1,1590 @@
+            // Trim or delete the mention
+            NSString *trimmedString = nil;
+            BOOL canTrim = [self mentionCanBeTrimmed:mentionAtDeleteLocation trimmedString:&trimmedString];
+            if (canTrim) {
+                // Trim mention to first word only
+                NSAssert([trimmedString length] > 0,
+                         @"Cannot trim a mention to zero length");
+                mentionAtDeleteLocation.mentionText = trimmedString;
+                [parentTextView transformTextAtRange:mentionRange
+                                     withTransformer:^NSAttributedString *(NSAttributedString *input) {
+                    return [input attributedSubstringFromRange:NSMakeRange(0, [trimmedString length])];
+                }];
+                // Move the cursor into position.
+                parentTextView.selectedRange = NSMakeRange(mentionRange.location + [trimmedString length],
+                                                           0);
+                // Notify the parent text view's external delegate that the text changed, since a mention was trimmed.
+                if (self.notifyTextViewDelegateOnMentionTrim
+                    && [externalDelegate respondsToSelector:@selector(textViewDidChange:)]) {
+                    [externalDelegate textViewDidChange:parentTextView];
+                }
+            }
+            else {
+                // Delete mention entirely
+                NSUInteger locationAfterDeletion = mentionRange.location;
+                [parentTextView transformTextAtRange:mentionRange
+                                     withTransformer:^NSAttributedString *(__unused NSAttributedString *input) {
+                    return (NSAttributedString *)nil;
+                }];
+                [self stripCustomAttributesFromTypingAttributes];
+                parentTextView.selectedRange = NSMakeRange(locationAfterDeletion, 0);
+
+                // Notify the parent text view's external delegate that the text changed, since a mention was deleted.
+                if (self.notifyTextViewDelegateOnMentionDeletion
+                    && [externalDelegate respondsToSelector:@selector(textViewDidChange:)]) {
+                    [externalDelegate textViewDidChange:parentTextView];
+                }
+            }
+            returnValue = NO;
+        }
+    }
+    return returnValue;
+}
+
+#pragma mark - Plug-in protocol
+
+- (void)dataReturnedWithEmptyResults:(BOOL)isEmptyResults
+         keystringEndsWithWhiteSpace:(BOOL)keystringEndsWithWhiteSpace {
+    [self.creationStateMachine dataReturnedWithEmptyResults:isEmptyResults keystringEndsWithWhiteSpace:keystringEndsWithWhiteSpace];
+}
+
+/*!
+ Returns whether a string is a single control character
+ */
+- (BOOL)isStringControlCharacter:(NSString *)text {
+    return text.length == 1 && [self.controlCharacterSet characterIsMember:[text characterAtIndex:0]];
+}
+
+// TODO: Remove text view from call
+// JIRA: XXXX
```

Comment:
what does this comment mean?

Reply:
added jira. And it's that we don't need the textView arg in the call after v2/v1 split is cleaned up